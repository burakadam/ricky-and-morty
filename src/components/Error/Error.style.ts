import styled from '@emotion/styled';

const Container = styled('div')`
  width: auto;
  padding: 20px;
  background-color: #b2102f;
  border-radius: 10px;
  text-align: center;
  color: white;
  display: table;
  margin: auto;
`;

export { Container };
