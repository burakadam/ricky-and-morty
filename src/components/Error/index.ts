export interface IErrorProps {
  message: string | undefined;
}

export * from './Error';
