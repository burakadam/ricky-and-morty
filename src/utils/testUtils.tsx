import type { PreloadedState } from '@reduxjs/toolkit';
import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { PropsWithChildren } from 'react';
import React from 'react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '@/store/store';
import { setupStore } from '@/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  // eslint-disable-next-line @typescript-eslint/no-shadow
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function Wrapper({ children }: { children: React.ReactElement }) {
  return <>{children}</>;
}

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';
export { customRender as render };
