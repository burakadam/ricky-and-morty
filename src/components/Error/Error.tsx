import { Typography } from '@mui/material';

import type { IErrorProps } from '.';
import * as S from './Error.style';

const Error = (props: IErrorProps) => {
  return (
    <S.Container>
      <Typography color="reset">{props.message || 'oops! Something went wrong'}</Typography>
    </S.Container>
  );
};

export { Error };
