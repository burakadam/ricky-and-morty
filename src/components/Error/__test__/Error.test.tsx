/* eslint-disable testing-library/no-node-access */
import { screen } from '@testing-library/react';

import { render } from '@/utils/testUtils';

import { Error } from '../Error';

const MESSAGE = 'Character not found!';

describe('error text component', () => {
  let container: HTMLElement;

  beforeEach(() => {
    // eslint-disable-next-line testing-library/no-render-in-setup
    ({ container } = render(<Error message={MESSAGE} />));
  });

  it('render component correctly', () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it('contain error message', () => {
    expect(screen.getByText(MESSAGE)).toBeInTheDocument();
  });

  it('contain default error message without message', () => {
    render(<Error />);
    expect(screen.getByText(MESSAGE)).toBeInTheDocument();
  });
});
