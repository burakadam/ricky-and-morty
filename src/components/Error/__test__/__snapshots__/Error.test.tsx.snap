// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error text component render component correctly 1`] = `
<div
  class="css-1tq42kr"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1b0s3yy-MuiTypography-root"
  >
    Character not found!
  </p>
</div>
`;
